tests/integration/suite/actors/healthz/deactivate-on-unhealthy.go

```diff
@@ -0,0 +1,158 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/slices"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	prochttp "github.com/dapr/dapr/tests/integration/framework/process/http"
+	"github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/framework/util"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(deactivateOnUnhealthy))
+}
+
+// deactivateOnUnhealthy ensures that all active actors are deactivated if the app is reported as unhealthy.
+type deactivateOnUnhealthy struct {
+	daprd               *daprd.Daprd
+	place               *placement.Placement
+	isHealthy           atomic.Bool
+	invokedActorsCh     chan string
+	deactivatedActorsCh chan string
+}
+
+func (h *deactivateOnUnhealthy) Setup(t *testing.T) []framework.Option {
+	h.isHealthy.Store(true)
+	h.invokedActorsCh = make(chan string, 2)
+	h.deactivatedActorsCh = make(chan string, 2)
+
+	r := chi.NewRouter()
+	r.Get("/dapr/config", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(`{"entities": ["myactortype"]}`))
+	})
+	r.Get("/healthz", func(w http.ResponseWriter, r *http.Request) {
+		if h.isHealthy.Load() {
+			w.WriteHeader(http.StatusOK)
+		} else {
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+	})
+	r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(`OK`))
+	})
+	r.Delete("/actors/{actorType}/{actorId}", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		act := fmt.Sprintf("%s/%s", chi.URLParam(r, "actorType"), chi.URLParam(r, "actorId"))
+		log.Printf("Received deactivation for actor %s", act)
+		h.deactivatedActorsCh <- act
+	})
+	r.Put("/actors/{actorType}/{actorId}/method/foo", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(`bar`))
+		act := fmt.Sprintf("%s/%s", chi.URLParam(r, "actorType"), chi.URLParam(r, "actorId"))
+		log.Printf("Received invocation for actor %s", act)
+		h.invokedActorsCh <- act
+	})
+
+	srv := prochttp.New(t, prochttp.WithHandler(r))
+	h.place = placement.New(t)
+	h.daprd = daprd.New(t, daprd.WithResourceFiles(`
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: mystore
+spec:
+  type: state.in-memory
+  version: v1
+  metadata:
+  - name: actorStateStore
+    value: true
+`),
+		daprd.WithPlacementAddresses("localhost:"+strconv.Itoa(h.place.Port())),
+		daprd.WithAppProtocol("http"),
+		daprd.WithAppPort(srv.Port()),
+	)
+
+	return []framework.Option{
+		framework.WithProcesses(h.place, srv, h.daprd),
+	}
+}
+
+func (h *deactivateOnUnhealthy) Run(t *testing.T, ctx context.Context) {
+	h.place.WaitUntilRunning(t, ctx)
+	h.daprd.WaitUntilRunning(t, ctx)
+
+	client := util.HTTPClient(t)
+
+	// Sleep a bit at the beginning as it takes time for the actor host to be registered
+	time.Sleep(500 * time.Millisecond)
```

Comment:
Besides `time.Sleep` being bad for tests generally as it artificially & arbitrarily slows things down- this doesn't look necessary as we are putting the invoke calls below in an eventually which will succeed as and when the placement is ready to accept requests, and whose start up time may likely change in future.